scheduler/workflow.go

```diff
@@ -315,8 +315,10 @@ func (s *schedulerWorkflow) Start(t *task) {
 	// Block until the job has been either run or skipped.
 	errors := t.manager.Work(j).Promise().Await()
 
-	if len(errors) != 0 {
-		t.RecordFailure(j.Errors())
+	if len(errors) != 0 && len(j.Errors()) > 0 {
```

Comment:
This line was changed to avoid panic's in the case where len(errors) != 0 but len(j.Errors()) == 0. Overwriting the new changes was not intentional.